task/sql-tasks.js

```diff
@@ -280,7 +462,27 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT
+	distinct
+            Customers.CompanyName,
+            Products.ProductName,
+            OrderDetails.UnitPrice as "PricePerItem"
+        FROM Customers 
+        JOIN Orders ON Customers.CustomerID=Orders.CustomerID
+        JOIN OrderDetails ON Orders.OrderID=OrderDetails.OrderID
+        JOIN Products ON OrderDetails.ProductID=Products.ProductID
+        WHERE OrderDetails.UnitPrice = (SELECT
+                    MAX(od.UnitPrice)
+                FROM Customers as c
+                JOIN Orders as o ON o.CustomerID=c.CustomerID
+                JOIN OrderDetails as od ON od.OrderID=o.OrderID
+                WHERE Customers.CompanyName=c.CompanyName

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.

Reply:
also try to use group by customer id and join subquery on unit price and customer id to speed up query